torch/reinforce/__init__.py

```diff
@@ -0,0 +1,104 @@
+"""
+The ``reinforce`` package contains sampling functions that return both the
+random samples and the log of the probability density function (pdf) or
+probability mass function (pmf) evaluated at each sample. These
+log-probabilities can be backpropagated through for policy gradient methods.
+
+Example::
+
+    probs = network(input)
+    action, log_prob = torch.reinforce.multinomial(probs)
+    loss = -log_prob * get_reward(env, action)
+    loss.backward()
+"""
+import math
+import torch
+
+
+__all__ = ['multinomial', 'bernoulli', 'normal']
+
+
+def multinomial(input, num_samples=1):
+    r"""
+    Samples from the multinomial distribution characterized by `input`. Returns
+    the samples and the log-probabilities of each returned sample.
+
+    See also: :func:`torch.multinomial`
+
+    .. note::
+
+        When `num_samples` is greater than one, this function always samples
+        with replacement.
```

Comment:
Why? `torch.multinomial` can return multiple samples without replacement, right?

Reply:
The probabilities are then conditional on the previously drawn samples. I'm not sure if that would affect policy-gradient methods.